backend/src/validation/membership.ts

```diff
@@ -78,3 +78,19 @@ export const DenyMembershipPermissionV1 = z.object({
     permissions: z.object({}).array()
   })
 });
+
+export const AddUserToWorkspaceV2 = z.object({
+  params: z.object({
+    workspaceId: z.string().trim()
+  }),
+  body: z.object({
+    members: z
+      .object({
+        orgMembershipId: z.string().trim(),
```

Comment:
I feel that it's strange that you need to know the `orgMembershipId` of the user you want to add to an organization.

This would work for dashboard purposes but in terms of exposing a REST API for users to in the future make these calls themselves, it makes it unnecessarily complicated; from a users POV it would make more sense to pass in the `userId` instead when making this call and have the backend perform required checks from there.

Reply:
Correct: This is to add to a workspace not organization

Have to disagree with that statement of complication rather i would say it made it easier for user. I'll tell you why.

Let's take the instance in which a user adds to a workspace. They will definitely need to fetch org membership of the user even to get user id. if ur fetching a org membership then u need to pass that. 

There is no api to get all user ids in an organization its all org membership filled with userids.

Also  by hierarchy of our design, org -> org membership -> workspace not in between user id which will end up lot more bugs in future

Reply:
Adding users to a workspace won't be easy either way because users will still need to do encryption to pass along the workspace key to invitees

Reply:
@maidul98 True — We will have to re-think this in the near future.